sdk/storage/azure-storage-blob/src/main/java/com/azure/android/storage/blob/StorageBlobClient.java

```diff
@@ -127,51 +127,35 @@ public ContainersListBlobFlatSegmentResponse getBlobsInPageWithRestResponse(Stri
-     * @param timeout               The timeout parameter is expressed in seconds. For more information, see
-     *                              &lt;a href="https://docs.microsoft.com/en-us/rest/api/storageservices/setting-timeouts-for-blob-service-operations"&gt;Setting Timeouts for Blob Service Operations.&lt;/a&gt;.
-     * @param version               Specifies the version of the operation to use for this request.
-     * @param blobRequestConditions Object that contains values which will restrict the successful operation of a
-     *                              variety of requests to the conditions present. These conditions are entirely
-     *                              optional.
-     * @param requestId             Provides a client-generated, opaque value with a 1 KB character limit that is
-     *                              recorded in the analytics logs when storage analytics logging is enabled.
-     * @param cpkInfo               Additional parameters for the operation.
-     * @param cancellationToken     The token to request cancellation.
+     * @param options               The optional parameter.
      * @return The response information returned from the server when downloading a blob.
      */
-    public BlobGetPropertiesResponse getBlobPropertiesWithRestResponse(String containerName,
-                                                                       String blobName,
-                                                                       String snapshot,
-                                                                       Integer timeout,
-                                                                       String version,
-                                                                       BlobRequestConditions blobRequestConditions,
-                                                                       String requestId,
-                                                                       CpkInfo cpkInfo,
-                                                                       CancellationToken cancellationToken) {
-        blobRequestConditions = blobRequestConditions == null ? new BlobRequestConditions() : blobRequestConditions;
-
-        return storageBlobServiceClient.getBlobPropertiesWithRestResponse(containerName,
+    public Response<BlobProperties> getBlobPropertiesWithRestResponse(String containerName,
+                                                            String blobName,
+                                                            GetBlobPropertiesOptions options) {
+        final BlobGetPropertiesResponse blobGetPropertiesResponse
+            = storageBlobServiceClient.getBlobPropertiesWithRestResponse(containerName,
             blobName,
-            snapshot,
-            timeout,
-            version,
-            blobRequestConditions.getLeaseId(),
-            requestId,
-            cpkInfo,
-            cancellationToken);
+            ClientUtil.toImplOptions(options));
```

Comment:
Will probably have to fix indentation here as well if we use the above suggestion.